experimental/CliffordAlgebras/src/CliffordAlgebras.jl

```diff
@@ -0,0 +1,579 @@
+export CliffordAlgebra,
+  CliffordAlgebraElem,
+  clifford_algebra,
+  even_coeff,
+  odd_coeff,
+  even_part,
+  odd_part,
+  centroid,
+  quadratic_discriminant,
+  disq
+
+################################################################################
+#
+#  Datatypes
+#
+################################################################################
+
+### Algebra ###
+mutable struct CliffordAlgebra{T,S} <: Hecke.AbstractAssociativeAlgebra{T}
+  base_ring::Ring
+  space::Hecke.QuadSpace{K,S} where {K}
+  gram::S
+  dim::Int
+  centroid::Any
+  disq::T
+  center::Any
+
+  #Return the Clifford algebra of the quadratic space 'qs' 
+  function CliffordAlgebra{T,S}(qs::Hecke.QuadSpace{K,S}) where {T,S,K}
+    gram = gram_matrix(qs)
+    return new{T,S}(base_ring(qs), qs, gram, 2^ncols(gram))
+  end
+end
+
+### Elements ###
+mutable struct CliffordAlgebraElem{T,S,P} <: Hecke.AbstractAssociativeAlgebraElem{T}
+  parent::P
+  coeffs::Vector{T}
+  even_coeffs::Vector{T}
+  odd_coeffs::Vector{T}
+
+  #Return the 0-element of the Clifford algebra C
+  function CliffordAlgebraElem{T,S,P}(C::CliffordAlgebra{T,S}) where {T,S,P}
+    newelt = new{T,S,CliffordAlgebra{T,S}}(C, fill(C.base_ring(), C.dim))
+    _set_even_odd_coeff!(newelt)
+    return newelt
+  end
+
+  CliffordAlgebraElem(C::CliffordAlgebra) =
+    CliffordAlgebraElem{elem_type(C.base_ring),typeof(C.gram),typeof(C)}(C)
+
+  #Return the element in the Clifford algebra C with coefficient vector coeff wrt. the canonical basis
+  function CliffordAlgebraElem{T,S,P}(
+    C::CliffordAlgebra{T,S}, coeff::Vector{R}
+  ) where {T,S,P,R<:FieldElem}
+    @req length(coeff) == C.dim "invalid length of coefficient vector"
+    newelt = new{T,S,CliffordAlgebra{T,S}}(C, coeff)
+    _set_even_odd_coeff!(newelt)
+    return newelt
+  end
+
+  CliffordAlgebraElem(C::CliffordAlgebra{T,S}, coeff::Vector{T}) where {T,S} =
+    CliffordAlgebraElem{elem_type(C.base_ring),typeof(C.gram),typeof(C)}(C, coeff)
+
+  function CliffordAlgebraElem(C::CliffordAlgebra{T,S}, coeff::Vector{R}) where {T,S,R}
+    K = C.base_ring
+    @req __can_convert_coeff(coeff, K) "entries of coefficient vector are not contained in $(K)"
+    return CliffordAlgebraElem{elem_type(C.base_ring),typeof(C.gram),typeof(C)}(
+      C, K.(coeff)
+    )
+  end
+end
+
+elem_type(::Type{CliffordAlgebra{T,S}}) where {T,S} =
+  CliffordAlgebraElem{T,S,CliffordAlgebra{T,S}}
+
+parent_type(::Type{CliffordAlgebraElem{T,S,P}}) where {T,S,P} = P
+
+base_ring_type(C::CliffordAlgebra{T,S}) where {T,S} = typeof(base_ring(C))
```

Comment:
`base_ring_type` should take `Type{CliffordAlgebra}` as its argument and should be computable at compile time. I think this could just be `parent_type(T)` here, but not sure

Reply:
Fixed in this [commit](https://github.com/oscar-system/Oscar.jl/pull/4370/commits/73f677cfe5c0362d5f9763d3d3b24a6e832e63a2)